python/gsw/main.py

```diff
@@ -0,0 +1,350 @@
+
+                    # Handles downlinks
+                    if email_subject.find("SBD Msg From Unit: ")==0:
+                        # Go through the email contents
+                        for part in msg.walk():
+                                        
+                            if part.get_content_maintype() == 'multipart':
+                                continue
+
+                            if part.get('Content-Disposition') is None:
+                                continue
+                                            
+                            # Get the body text of the email and look for the MOMSN/MTMSN number
+                            if part.get_content_type() == "text/plain":
+                                email_body = part.get_payload(decode=True).decode('utf8')
+                                for line in email_body.splitlines():
+                                    if line.find("MOMSN")!=-1:
+                                        self.momsn=int(line[7:])
+                                    if line.find("MTMSN")!=-1:
+                                        self.confirmation_mtmsn=int(line[7:])
+                            
+                            #Verify whether or not radioSession can send uplinks
+                            if self.confirmation_mtmsn != self.mtmsn:
+                                #stop radio session from sending any more uplinks
+                                self.send_uplinks=False
+                            else:
+                                #allow radio session to send more uplinks
+                                self.send_uplinks=True
+
+                            # Check if there is an email attachment
+                            if part.get_filename() is not None:
+                                # Get data from email attachment
+                                attachmentContents=part.get_payload(decode=True).decode('utf8')
+                                statefield_report=self.process_downlink_packet(attachmentContents)
+                                return statefield_report
+                        
+                    #if we have not recieved a downlink, return None
+                    return None
+
+    def post_to_es(self):
+        '''
+        Check for the most recent email from iridium. 
+        There are two conditions under which we would create an Iridium report: 
+        1) When we recieve a statefield report
+        2) when we recieve an uplink confirmation.
+        - If there is a statefield report, index the statefield report and 
+        create and index an iridium report.
+        - If there is not a statefield report but we recently recieved an uplink
+        confirmation, then create and index an iridium report.
+        '''
+        # Connect to elasticsearch
+        es=Elasticsearch([{'host':self.es_server,'port':self.es_port}])
+
+        while self.run_email_thread==True:
```

Comment:
This while loop will currently eat up a lot of CPU bandwidth because there's no waiting at the end of the loop. Consider adding a 1 second delay to the loop so that other computer processes can run in the meantime.

Re: this, can you quantify how many resources this takes up? RAM, CPU, etc. This is important so we can scale down our Virtual Machine if possible

Reply:
Marking as "unresolved" because I'm interested in these numbers

Reply:
If the CPU numbers are high (I suspect they are, given how much your fan spins up during work sessions...:P), I honestly think we can let the delay be as large as 10 seconds. I think 10 seconds is a sweetspot because we won't see a noticeable reduction in resource utilization by increasing beyond 10 seconds, and emails are only going to come once every few hours, so running this super frequently isn't a good idea